pkg/network/domainspec/generators.go

```diff
@@ -299,51 +307,63 @@ func (b *PasstLibvirtSpecGenerator) Generate() error {
 		}
 	}()
 
-	err = cmd.Start()
-	if err != nil {
-		log.Log.Reason(err).Error("failed to start passt")
-		return err
-	}
+	return nil
+}
 
-	err = cmd.Wait()
-	if err != nil {
-		log.Log.Reason(err).Error("failed waiting for passt going to background")
-		return err
+const (
+	protoTCP = "tcp"
+	protoUDP = "udp"
+)
+
+func (b *PasstLibvirtSpecGenerator) generateInterface(iface *api.Interface) *api.Interface {
+	ifaceCopy := iface.DeepCopy()
+
+	var mac *api.MAC
+	if b.vmiSpecIface.MacAddress != "" {
+		mac = &api.MAC{MAC: b.vmiSpecIface.MacAddress}
 	}
 
-	return nil
+	ifaceCopy.Type = ifaceTypeUser
+	ifaceCopy.Source = api.InterfaceSource{Device: namescheme.PrimaryPodInterfaceName}
```

Comment:
This is assuming the primary pod interface name. Isn't this name somehow available without this assumption? It opens up a lot of questions at this low level code.

Reply:
I agree with Eddy, `b.podInterfaceName` can be used instead.

Reply:
> `b.podInterfaceName` can be used instead.

There is a need to add it to the passt structure.
The act of detecting the correct interface name already exists for the other bindings and can be used with passt as well.

What I do not understand is: Why it was not needed before?

Reply:
It wasn't needed since qrap chose automatically the interface of the default gateway.

Reply:
I cherry-picked the commit that fix this from #10457